build_test.go

```diff
@@ -0,0 +1,287 @@
+		Expect(layer.SharedEnv["PATH.delim"]).To(Equal(":"))
+		Expect(layer.SharedEnv["PATH.append"]).To(Equal(filepath.Join(layersDir, "vsdbg")))
+
+		Expect(layer.BuildEnv).To(BeEmpty())
+		Expect(layer.LaunchEnv).To(BeEmpty())
+		Expect(layer.ProcessLaunchEnv).To(BeEmpty())
+
+		Expect(layer.Build).To(BeFalse())
+		Expect(layer.Launch).To(BeFalse())
+		Expect(layer.Cache).To(BeFalse())
+
+		Expect(layer.Metadata).To(HaveLen(1))
+		Expect(layer.Metadata["dependency_sha"]).To(Equal("vsdbg-dependency-sha"))
+
+		Expect(layer.SBOM.Formats()).To(Equal([]packit.SBOMFormat{
+			{
+				Extension: sbom.Format(sbom.CycloneDXFormat).Extension(),
+				Content:   sbom.NewFormattedReader(sbom.SBOM{}, sbom.CycloneDXFormat),
+			},
+			{
+				Extension: sbom.Format(sbom.SPDXFormat).Extension(),
+				Content:   sbom.NewFormattedReader(sbom.SBOM{}, sbom.SPDXFormat),
+			},
+		}))
+
+		Expect(dependencyManager.ResolveCall.Receives.Path).To(Equal(filepath.Join(cnbDir, "buildpack.toml")))
+		Expect(dependencyManager.ResolveCall.Receives.Id).To(Equal("vsdbg"))
+		Expect(dependencyManager.ResolveCall.Receives.Version).To(Equal(""))
+		Expect(dependencyManager.ResolveCall.Receives.Stack).To(Equal("some-stack"))
+
+		Expect(dependencyManager.DeliverCall.Receives.Dependency).To(Equal(postal.Dependency{
+			ID:      "vsdbg",
+			Name:    "vsdbg-dependency-name",
+			SHA256:  "vsdbg-dependency-sha",
+			Stacks:  []string{"some-stack"},
+			URI:     "vsdbg-dependency-uri",
+			Version: "vsdbg-dependency-version",
+		}))
+
+		Expect(dependencyManager.DeliverCall.Receives.CnbPath).To(Equal(cnbDir))
+		Expect(dependencyManager.DeliverCall.Receives.DestinationPath).To(Equal(filepath.Join(layersDir, "vsdbg")))
+		Expect(dependencyManager.DeliverCall.Receives.PlatformPath).To(Equal("platform"))
+
+		Expect(sbomGenerator.GenerateCall.Receives.Dir).To(Equal(filepath.Join(layersDir, "vsdbg")))
+
+		Expect(buffer.String()).To(ContainSubstring("Some Buildpack some-version"))
+		Expect(buffer.String()).To(ContainSubstring("Executing build process"))
+		Expect(buffer.String()).To(ContainSubstring("Installing Visual Studio Debugger"))
+
+		info, err := os.Stat(filepath.Join(layersDir, "vsdbg", "vsdbg"))
+		Expect(err).NotTo(HaveOccurred())
+		Expect(info.Mode().String()).To(Equal("-rwx--x---"))
+	})
+
+	context("when build plan entries require vsdbg at build/launch", func() {
+		it.Before(func() {
+			buildContext.Plan.Entries[0].Metadata = make(map[string]interface{})
+			buildContext.Plan.Entries[0].Metadata["build"] = true
+			buildContext.Plan.Entries[0].Metadata["launch"] = true
+		})
+
+		it("makes the layer available at the right times", func() {
+			result, err := build(buildContext)
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(result.Layers).To(HaveLen(1))
+			layer := result.Layers[0]
+
+			Expect(layer.Name).To(Equal("vsdbg"))
+
+			Expect(layer.Build).To(BeTrue())
+			Expect(layer.Launch).To(BeTrue())
+			Expect(layer.Cache).To(BeTrue())
+
+		})
+	})
```

Comment:
I think it'd also be good to unit test that the layer is set up as expected when the cached layer is reused.